x-pack/legacy/plugins/fleet/server/libs/agent.ts

```diff
@@ -199,16 +199,20 @@ export class AgentLib {
    */
   public async checkin(
     user: FrameworkUser,
-    agentId: string,
     events: AgentEvent[],
     localMetadata?: any
-  ): Promise<{ actions: AgentAction[]; policy: FullPolicyFile }> {
-    const agent = await this.agentsRepository.getById(user, agentId);
+  ): Promise<{ actions: AgentAction[]; policy: FullPolicyFile | null }> {
+    const res = await this.apiKeys.verifyAccessApiKey(user);
```

Comment:
Rather than saving an ID of the api key on the agent we stored a hash of it. We could just hash it here from the header rather then doing a second query. Then call getByAccessApiKeyId (renaming it to API Key hash or something) with the requesting user vs internal.

Reply:
Looking for feedback here, not saying this is what we MUST do

Reply:
the api key id is stored and readable from the header so we do not need to do an extra query, we should store the id to be able to invalidate the key later, so it's probably easier to store only one field.

Elastic api key works like this `Authorization: ApiKey {Base64({keyId}:{key})}`